src/forms/Y2021/tests/taxRates.test.ts

```diff
@@ -0,0 +1,253 @@
+import { FilingStatus } from 'ustaxes/core/data'
+import { CURRENT_YEAR } from '../data/federal'
+import { computeOrdinaryTax } from '../irsForms/TaxTable'
+
+describe('Tax rates', () => {
+  it('test should be updated for new year', async () => {
+    // WARNING! Do not just chagne the year. Also update the expected tax amounts below!
+    expect(CURRENT_YEAR).toEqual(2021)
+  })
+  it('ordinary taxes for single status should be correct', async () => {
+    expect(Math.round(computeOrdinaryTax(FilingStatus.S, 5))).toEqual(1)
```

Comment:
This could be handled a bit better I think, no reason you can't just have a CSV file loaded, loop over it and assert `f(x) = y` for each row in the file for example, or even doing a `.foreach` just on a JS array loaded above here would be fine.

Reply:
There's also a property based testing approach that could work. You could have the tax table you linked imported directly as a CSV, and then generate arbitrary incomes, and assert for income bracket I and tax T, 

```
I_0 < x <= I_1, T_0 < f(x) <= T_1`
```

Reply:
I updated it to pull in a CSV, then a few points in every tax table range.